pkg/doc/signature/jsonld/processor.go

```diff
@@ -26,23 +26,31 @@ var logger = log.New("aries-framework/json-ld-processor")
 // normalizeOpts holds options for canonicalization of JSON LD docs
 type normalizeOpts struct {
 	removeInvalidRDF bool
-	removeRDFMatches []string
+	documentLoader   ld.DocumentLoader
+	extraContexts    []string
 }
 
-// CanonicalizationOpts are the options for canonicalization of JSON LD docs
-type CanonicalizationOpts func(opts *normalizeOpts)
+// ProcessorOpts are the options for JSON LD operations on docs (like canonicalization or compacting).
+type ProcessorOpts func(opts *normalizeOpts)
 
 // WithRemoveAllInvalidRDF option for removing all invalid RDF dataset from normalize document
-func WithRemoveAllInvalidRDF() CanonicalizationOpts {
+func WithRemoveAllInvalidRDF() ProcessorOpts {
 	return func(opts *normalizeOpts) {
 		opts.removeInvalidRDF = true
 	}
 }
 
-// WithRemoveInvalidRDF option for removing RDFs containing any one of the given matches
-func WithRemoveInvalidRDF(matches ...string) CanonicalizationOpts {
+// WithDocumentLoader option is for passing custom JSON-LD document loader.
+func WithDocumentLoader(loader ld.DocumentLoader) ProcessorOpts {
 	return func(opts *normalizeOpts) {
-		opts.removeRDFMatches = matches
+		opts.documentLoader = loader
+	}
+}
+
+// WithExternalContext option is for definition of external context when doing JSON-LD operations.
+func WithExternalContext(context ...string) ProcessorOpts {
+	return func(opts *normalizeOpts) {
+		opts.extraContexts = context
```

Comment:
for consistency, would be good to replace-all on opts.extraContexts

Reply:
ah, forgot to rename that.